lib/broccoli/ember-app.js

```diff
@@ -985,11 +1017,17 @@ class EmberApp {
       annotation: 'TreeMerger (app)',
     });
 
-    return new Funnel(mergedApp, {
+    let appTree = new Funnel(mergedApp, {
       srcDir: '/',
       destDir: this.name,
       annotation: 'ProcessedAppTree',
     });
+
+    if (experiments.DELAYED_TRANSPILATION && isPackageHookSupplied) {
```

Comment:
shouldn't we check if packager experiment is enabled or not rather than checking for `package` property?